src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -1987,13 +1968,29 @@ private object GetMessageJson(JsonRpcMessage message)
             }
         }
 
-#pragma warning disable AvoidAsyncSuffix // Avoid Async suffix
-        private ValueTask TransmitAsync(JsonRpcMessage message, CancellationToken cancellationToken)
-#pragma warning restore AvoidAsyncSuffix // Avoid Async suffix
+        private async ValueTask TransmitAsync(JsonRpcMessage message, CancellationToken cancellationToken)
         {
-            this.TraceMessageSent(message);
+            try
+            {
+                this.TraceMessageSent(message);
+                await this.MessageHandler.WriteAsync(message, cancellationToken).ConfigureAwait(false);
+            }
+            catch (OperationCanceledException)
+            {
+            }
```

Comment:
please rethrow all exceptions you catch. The callers are expecting to be able to discern between success and failure based on the exceptions thrown from this method.